src/server/game/AuctionHouse/AuctionHouseListing.cpp

```diff
@@ -25,63 +25,54 @@
 #include "Player.h"
 #include "GameTime.h"
 
-#include <boost/thread/locks.hpp>
-#include <boost/thread/mutex.hpp>
+std::list<AuctionListingEvent*> AuctionHouseListing::_requestsNew;
+std::list<AuctionListingEvent*> AuctionHouseListing::_requests;
 
-#include <condition_variable>
-#include <vector>
-#include <unordered_map>
+std::mutex AuctionHouseListing::_listingLock;
+std::atomic<bool> AuctionHouseListing::_auctionHouseListingAllowed = false;
 
 namespace
 {
-    std::list<AuctionListingEvent*> _requestsNew;
-    std::list<AuctionListingEvent*> _requests;
-
-    std::mutex _listingLock;
-    std::condition_variable _allowedCondition;
-    std::atomic<bool> _auctionHouseListingAllowed;
-
     // Map of throttled players for GetAll, and throttle expiry time
     // Stored here, rather than player object to maintain persistence after logout
-    std::unordered_map<ObjectGuid, time_t> GetAllThrottleMap;
+    std::unordered_map<ObjectGuid, time_t> _getAllThrottleMap;
 }
 
-void AuctionHouseListing::AuctionHouseListingThread()
+void AuctionHouseListing::Update()
 {
-    TC_LOG_INFO("misc", "Auction House Listing Thread starting");
+    if (!IsListingAllowed())
+        return;
 
-    while (!World::IsStopped())
     {
         std::unique_lock<std::mutex> lock(_listingLock);
-        while (!IsListingAllowed())
-            _allowedCondition.wait(lock);
 
         // process new requests
-        {
-            if (!_requestsNew.empty())
-                _requests.splice(_requests.end(), _requestsNew);
-        }
+        if (!_requestsNew.empty())
+            _requests.splice(_requests.end(), _requestsNew);
 
-        if (!_requests.empty())
+        for (auto itr = _requests.begin(); itr != _requests.end();)
         {
-            for (auto itr = _requests.begin(); itr != _requests.end();)
+            if ((*itr)->Execute())
             {
-                if ((*itr)->Execute())
-                {
-                    delete (*itr);
-                    itr = _requests.erase(itr);
-                }
-                else
-                    ++itr;
-
-                // main thread may be waiting for us
-                if (!IsListingAllowed())
-                    break;
+                delete (*itr);
+                itr = _requests.erase(itr);
+                break;
             }
+            else
+                ++itr;
         }
     }
+}
 
-    TC_LOG_INFO("misc", "Auction House Listing thread exiting without problems.");
+void AuctionHouseListing::AuctionHouseListingHandler(boost::asio::deadline_timer* updateAuctionTimer, boost::system::error_code const& error)
```

Comment:
Nah, just doesn't execute if set (like all the other handlers hooked to _ioService)